shared/src/question-generators/graph-algorithms/dijkstraTable.ts

```diff
@@ -0,0 +1,440 @@
+    invalid_set: "Invalid node set.",
+    invalid_dist: "Invalid distance.",
+    invalid_pred: "Invalid predecessor.",
+    inconsistency_error:
+      'Input "{{input}}" is inconsistent with the value "{{previous}}" in the previous row.',
+  },
+  de: {
+    name: "Dijkstra Tabelle",
+    description:
+      "Nutze den Dijkstra-Algorithmus um die Tabelle mit den Entfernungen zwischen Knoten auszufüllen",
+    tableQuestion:
+      "Nutze den Dijkstra-Algorithmus um im folgenden ungerichteten Graphen mit Kantengewichten die kürzesten Wege vom Knoten $A$ zu allen anderen Knoten u bestimmen.",
+    tablePrompt:
+      "Fülle dazu die folgende Tabelle aus. Hierbei entspricht jede Zeile einem Schritt des Algorithmus. Die Spalte $d(n)$ soll den aktuellen Abstand, die Spalte $p(n)$ den Vorgänger für den Knoten $n$ im Elternarray enthalten. Gib für die Menge $S$ an, welcher Knoten in jedem Schritt hinzugefügt wird. Es ist nicht notwendig Zellen auszufüllen, die sich nicht von der vorhergehenden Zeile unterscheiden.",
+    invalid_set: "keine gültige Knotenmenge.",
+    invalid_dist: "keine gültige Distanz.",
+    invalid_pred: "kein gültiger Abstand.",
+    inconsistency_error:
+      'Eingabe "{{input}}" stimmt nicht mit dem Wert "{{previous}}" in der vorherigen Zeile überein.',
+  },
+}
+
+export const DijkstraTableGenerator: QuestionGenerator = {
+  id: "dijktable",
+  name: tFunctional(translations, "name"),
+  description: tFunctional(translations, "description"),
+  tags: ["graph", "dijkstra", "table"],
+  languages: ["en", "de"],
+  expectedParameters: [
+    {
+      name: "size",
+      type: "integer",
+      min: 2,
+      max: 8,
+    },
+  ],
+
+  generate: (lang: Language, parameters, seed) => {
+    const random = new Random(seed)
+
+    const { graph, steps } = generateRandomGraphWithSteps(random, parameters.size as number)
+
+    const permalink = serializeGeneratorCall({
+      generator: DijkstraTableGenerator,
+      lang,
+      parameters,
+      seed,
+    })
+
+    const table = getDijkstraInputTable(steps, graph)
+    const feedback = getFeedbackFunction(steps, graph)
+    const checkFormat = getCheckFormatFunction(lang, graph)
+
+    console.log("debug:\n", cheatForDebugging(steps, graph))
+
+    const question: MultiFreeTextQuestion = {
+      type: "MultiFreeTextQuestion",
+      path: permalink,
+      name: t(translations, lang, "name"),
+      fillOutAll: false,
+      text: `${t(translations, lang, "tableQuestion")}\n${graph.toMarkdown()}\n${t(
+        translations,
+        lang,
+        "tablePrompt",
+      )}\n\n${table}`,
+      feedback,
+      checkFormat,
+    }
+
+    return { question }
+  },
+}
+
+function cheatForDebugging(
```

Comment:
Is this function being used, or is its sole purpose debugging? If it is being used, I believe we should either delete it or comment it out.